test/transport-stream.test.js

```diff
@@ -1,14 +1,15 @@
 'use strict'
 
-const { test } = require('tap')
+const { test } = require('node:test')
+const { mock } = require('cjs-mock')
```

Comment:
I installed `cjs-mock` because mocking in the node test runner is experimental.

Reply:
I pushed a commit. I prefer dependency injection for such things.

Reply:
LGTM!